docs/en/integrations/data-ingestion/data-formats/parquet-arrow-avro-orc.md

```diff
@@ -0,0 +1,309 @@
+
+```sql
+INSERT INTO sometable
+FROM INFILE 'data.arrow'
+FORMAT Arrow
+```
+
+Exporting to Arrow file works the same way:
+
+```sql
+SELECT * FROM sometable
+INTO OUTFILE 'export.arrow'
+FORMAT Arrow
+```
+
+Also, check [data types matching](/docs/en/interfaces/formats.md/#data-types-matching-arrow) to know if any should be converted manually.
+
+### Arrow data streaming
+
+The [ArrowStream](/docs/en/interfaces/formats.md/#data-format-arrow-stream) format can be used to work with Arrow streaming (used for in-memory processing). ClickHouse can read and write Arrow streams.
+
+To demonstrate how ClickHouse can stream Arrow data, let's pipe it to the following python script (it reads input stream in Arrow streaming format and outputs the result as a Pandas table):
+
+```python
+import sys, pyarrow as pa
+
+with pa.ipc.open_stream(sys.stdin.buffer) as reader:
+  print(reader.read_pandas())
+```
+
+Now we can stream data from ClickHouse by piping its output to the script:
+
+```bash
+clickhouse-client -q "SELECT path, hits FROM some_data LIMIT 3 FORMAT ArrowStream" | python3 arrow.py
+
+                           path  hits
+0       b'Akiba_Hebrew_Academy'   241
+1           b'Aegithina_tiphia'    34
+2  b'1971-72_Utah_Stars_season'     1
+```
+
+ClickHouse can read Arrow streams as well using the same ArrowStream format:
+
+```sql
+arrow-stream | clickhouse-client -q "INSERT INTO sometable FORMAT ArrowStream"
+```
+
+We've used `arrow-stream` as a possible source of Arrow streaming data.
+
+## Importing and exporting ORC data
+
+[Apache ORC](https://orc.apache.org/) format is a columnar storage format typically used for Hadoop. ClickHouse supports importing as well as exporting [Orc data](assets/data.orc) using [ORC format](/docs/en/interfaces/formats.md/#data-format-orc):
+
+```sql!
+SELECT *
+FROM sometable
+INTO OUTFILE 'data.orc'
+FORMAT ORC;
+
+INSERT INTO sometable
+FROM INFILE 'data.orc'
+FORMAT ORC;
+```
+
+Also, check [data types matching](/docs/en/interfaces/formats.md/#data-types-matching-orc) as well as [additional settings](/docs/en/interfaces/formats.md/#parquet-format-settings) to tune export and import.
+
+## Further reading
+
+ClickHouse introduces support for many formats, both text, and binary, to cover various scenarios and platforms. Explore more formats and ways to work with them in the following articles:
+
+- [CSV and TSV formats](csv-tsv.md)
+- **Parquet, Avro, Arrow and ORC**
+- [JSON formats](json.md)
+- [Regex and templates](templates-regex.md)
+- [Native and binary formats](binary.md)
+- [SQL formats](sql.md)
+
+And also check [clickhouse-local](https://clickhouse.com/blog/extracting-converting-querying-local-files-with-sql-clickhouse-local) - a portable full-featured tool to work on local/remote files without the need for Clickhouse server.
```

Comment:
ohh can we put this on the bottom of all these docs @mrcrypster ?

Reply:
Sounds good, added!